js/forms.js

```diff
@@ -310,76 +310,185 @@
+              matchText = $el.text().slice(matchStart, matchEnd + 1),
+              afterMatch = $el.text().slice(matchEnd + 1);
+          $el.html("<span>" + beforeMatch + "<span class='highlight'>" + matchText + "</span>" + afterMatch + "</span>");
+          if (img.length) {
+            $el.prepend(img);
+          }
+          // Other, in case of remote, can be any other property to set as $.data properties
+          if( other ) {
+            $el.data(other);
+          }
+        };
 
-          // Reset current element position
-          var resetCurrentElement = function() {
-            activeIndex = -1;
-            $autocomplete.find('.active').removeClass('active');
+        // Reset current element position
+        var resetCurrentElement = function() {
+          activeIndex = -1;
+          $autocomplete.find('.active').removeClass('active');
+        }
+
+        // Remove autocomplete elements
+        var removeAutocomplete = function() {
+          $autocomplete.empty();
+          resetCurrentElement();
+          oldVal = undefined;
+        };
+
+        // Build autocompleteOption
+        var buildAutocompleteOption = function(val, string, img, other) {
+          var autocompleteOption = $('<li></li>');
+
+          if (!!img) {
+            autocompleteOption.append('<img src="'+ img +'" class="right circle"><span>'+ string +'</span>');
+          } else {
+            autocompleteOption.append('<span>'+ string +'</span>');
           }
 
-          // Remove autocomplete elements
-          var removeAutocomplete = function() {
-            $autocomplete.empty();
-            resetCurrentElement();
-            oldVal = undefined;
-          };
+          $autocomplete.append(autocompleteOption);
+          highlight(val, autocompleteOption, other);
+        }
 
-          $input.off('blur.autocomplete').on('blur.autocomplete', function() {
-            removeAutocomplete();
-          });
+        // Perform classic search
+        var performSearch = function(val) {
+          var count = 0;
+
+          for(var key in data) {
+            if (data.hasOwnProperty(key) &&
+                key.toLowerCase().indexOf(val) !== -1) {
+              // Break if past limit
+              if (count >= options.limit) {
+                break;
+              }
+              
+              buildAutocompleteOption(val, key, data[key]);          
+              count++;
+            }
+          }
+        }
+
+        // Debounced, remote search
+        var performRemoteSearch = debounce(function() {
+          var val = $input.val().toLowerCase();
+          
+          // If we already have an xhr running, stop it
+          if(xhr) xhr.abort();
+
+          xhr = $
+            .ajax({
+              type: 'GET',
+              url: options.url,
+              data: { val: val }
+            })
+            .always(function() { xhr = null })
+            .fail(function(a,b,c) { console.log(a,b,c) })
```

Comment:
Displaying a toast instead of silence death ?

Reply:
Well, the user doesn't have to know that the Ajax call failed. The console is here for the dev during the dev process. After that, if the request fail, the behavior is the same than when there's no result since before my PR: nothing happend.